mava/systems/tf/mad4pg/training.py

```diff
@@ -661,19 +661,18 @@ def _forward(self, inputs: reverb.ReplaySample) -> None:
                 # Cast the additional discount to match
                 # the environment discount dtype.
                 agent_discount = discounts[agent]
-                discount = tf.cast(self._discount, dtype=agent_discount.dtype)
+                # discount = tf.cast(self._discount, dtype=agent_discount.dtype)
 
                 # Critic loss.
                 critic_loss = recurrent_n_step_critic_loss(
-                    q_values,
-                    target_q_values,
-                    rewards[agent],
-                    discount * agent_discount,
+                    q_values=q_values,
+                    target_q_values=target_q_values,
+                    rewards=rewards[agent],
+                    discount=self._discount,
+                    end_of_episode=agent_discount,
                     bootstrap_n=self._bootstrap_n,
-                    loss_fn=losses.categorical,
+                    # loss_fn=losses.categorical,
                 )
-                self.critic_losses[agent] = tf.reduce_mean(critic_loss, axis=0)
-
                 # Actor learning.
                 obs_agent_feed = target_obs_trans[agent]
                 # TODO (dries): Why is there an extra tuple?
```

Comment:
Is this still relevant here? I think a similar todo existed somewhere else and was removed?

Reply:
Removed.